cmd/juju/application/deployer/charm.go

```diff
@@ -578,6 +586,36 @@ func (c *repositoryCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI DeployerA
 	return c.deploy(ctx, deployAPI)
 }
 
+func (c *repositoryCharm) uploadExistingPendingResources(
+	appName string,
+	pendingResources []application.PendingResourceUpload,
+	api DeployerAPI) error {
+
+	resourceApiClient, err := resources.NewClient(api)
```

Comment:
Check to see if there are any pending resources to upload before opening the client. Saves juju cycles. Especially when the majority of charms won't have local resources to upload.